copy/copy.go

```diff
@@ -465,12 +459,24 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 		}
 	}
 
+	// Check if the updates or a type conversion meaningfully changed the list of images.
+	updatedManifestList, err := list.Serialize()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Error encoding updated manifest list (%q: %#v)", list.MIMEType(), list.Instances())
+	}
+	originalManifestList, err := originalList.Serialize()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Error encoding original manifest list for comparison (%q: %#v)", originalList.MIMEType(), originalList.Instances())
+	}
+	listIsModified := !bytes.Equal(updatedManifestList, originalManifestList)
+
 	// If we can't use the original value, but we have to change it, flag an error.
 	if listIsModified {
-		manifestList, err = list.Serialize()
-		if err != nil {
-			return nil, "", errors.Wrapf(err, "Error encoding updated manifest list (%q: %#v)", list.MIMEType(), list.Instances())
+		canModifyManifestList := (len(sigs) == 0)
```

Comment:
This sanity check is nice.

`canModifyManifestList` is computed earlier as well, can it be done only once to make sure the logic doesn’t diverge?

Reply:
Switched to reusing a `canModifyManifestList`.